indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java

```diff
@@ -2092,4 +2109,47 @@ protected abstract void possiblyResetDataSourceMetadata(
   protected abstract boolean isEndOffsetExclusive();
 
   protected abstract TypeReference<List<SequenceMetadata<PartitionIdType, SequenceOffsetType>>> getSequenceMetadataTypeReference();
+
+
+  private void addTaskDurationToMinMaxTimes()
+  {
+    if (minMessageTime.isPresent()) {
+      minMessageTime = Optional.of(minMessageTime.get()
```

Comment:
Is it ok that intelliJ is warning that this is an unsafe update of a volatile variable?

Reply:
Yeah its ok.

Its complaining because we are using the same variable to update the value but we only do it once every taskDuration. If it was done more frequently we should have used atomic references.